metrics/opentelemetry.go

```diff
@@ -8,22 +8,46 @@ import (
+	RetrievalDealSize                        = stats.Int64("retrieval_deal_size_bytes", "The size in bytes of a retrieval deal with a storage provider", stats.UnitDimensionless)
+	RetrievalDealSuccessCount                = stats.Int64("retrieval_deal_success_total", "The number of successful retrieval deals with storage providers", stats.UnitDimensionless)
+	RetrievalRequestCount                    = stats.Int64("retrieval_request_total", "The number of retrieval deals initiated with storage providers", stats.UnitDimensionless)
+	RetrievalResponseCount                   = stats.Int64("retrieval_response_total", "The number of retrieval responses", stats.UnitDimensionless)
+	RetrievalErrorPaychCount                 = stats.Int64("retrieval_error_paych_total", "The number of retrieval errors for 'failed to get payment channel'", stats.UnitDimensionless)
+	RetrievalErrorRejectedCount              = stats.Int64("retrieval_error_rejected_total", "The number of retrieval errors for 'response rejected'", stats.UnitDimensionless)
+	RetrievalErrorTooManyCount               = stats.Int64("retrieval_error_toomany_total", "The number of retrieval errors for 'Too many retrieval deals received'", stats.UnitDimensionless)
+	RetrievalErrorACLCount                   = stats.Int64("retrieval_error_acl_total", "The number of retrieval errors for 'Access Control'", stats.UnitDimensionless)
+	RetrievalErrorMaintenanceCount           = stats.Int64("retrieval_error_maintenance_total", "The number of retrieval errors for 'Under maintenance, retry later'", stats.UnitDimensionless)
+	RetrievalErrorNoOnlineCount              = stats.Int64("retrieval_error_noonline_total", "The number of retrieval errors for 'miner is not accepting online retrieval deals'", stats.UnitDimensionless)
+	RetrievalErrorUnconfirmedCount           = stats.Int64("retrieval_error_unconfirmed_total", "The number of retrieval errors for 'unconfirmed block transfer'", stats.UnitDimensionless)
+	RetrievalErrorOtherCount                 = stats.Int64("retrieval_error_other_total", "The number of retrieval errors with uncategorized causes", stats.UnitDimensionless)
 )
 
+// ErrorMetricMatches is a mapping of retrieval error message substrings (i.e.
+// that can be matched against error messages) and metrics to report for that
+// error.
+var ErrorMetricMatches = map[string]*stats.Int64Measure{
+	"failed to get payment channel":                 RetrievalErrorPaychCount,
+	"response rejected":                             RetrievalErrorRejectedCount,
+	"Too many retrieval deals received":             RetrievalErrorTooManyCount,
+	"Access Control":                                RetrievalErrorACLCount,
+	"Under maintenance, retry later":                RetrievalErrorMaintenanceCount,
+	"miner is not accepting online retrieval deals": RetrievalErrorNoOnlineCount,
+	"unconfirmed block transfer":                    RetrievalErrorUnconfirmedCount,
+}
```

Comment:
I really like this, but question whether or not it should live here? Not really a deal breaker though.

Reply:
🤷 it's directly connected to the specific error names

hopefully it doesn't live here permanently because we've squashed most of these and come to view the rest as reasonable and not worth monitoring!